File: ouvidoria.py
lista = []
opcao = 10
while opcao != 5:

    # Listar reclamacoes
    print()
    print("1) Listar reclamações")
    print("2) Adcionar nova reclamação")
    print("3) Remover uma reclamação")
    print("4) Pesquisar uma reclamação por código")
    print("5) Sair")
    print()
    opcao = int(input("Bem vindo à ouvidoria. Escolha o número da sua opção: "))

    # Listar reclamações
    if opcao == 1:
        if len(lista) == 0:
            print()
            print("Nenhuma reclamação cadastrada no sistema")
        else:
            for index, item in enumerate(lista):
                print(f"{index+1}) {item}")

    # Adcionar novas reclamações
    elif opcao == 2:
        reclame = input("Digite sua reclamação: ")
        lista.append(reclame)
        print()
        print("Reclamação cadastrada com sucesso")
        print(f"O código da reclamação é {len(lista)}")

    # Remover uma reclamação
    elif opcao == 3:
        for index, item in enumerate(lista):
            print(f"{index+1}) {item}")
        remocao = int(input("Digite o número da reclamação a ser removida: "))
        if remocao <= 0 or remocao > len(lista):
            print()
            print("Código inválido")
        else:
            lista.pop(remocao - 1)

    # Pesquisar uma reclamação por código
    elif opcao == 4:
        pesquisa = int(input("Digite o número da reclamação a ser pesquisada: "))
        if pesquisa <= 0 or pesquisa > len(lista):
            print()
            print("Código inválido")
        else:
            print()
            print(lista[pesquisa - 1])

    # Opção inválida e saída
    elif opcao != 5:
        print("Opção inválida")

print("Adeus")

File: ouvidoria_java/Ouvidoria.java
package application;

import java.util.Scanner;

public class Main {

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		String[] ocorrencias = new String[5];

		int opcao = 9;

		while (opcao != 3) {
			menu();
			System.out.print("Digite uma opção: ");
			opcao = sc.nextInt();
			sc.nextLine();
			switch (opcao) {
			case 1: {
				System.out.println("Inserir reclamações");
				for (int x = 0; x < ocorrencias.length; x++) {
					System.out.println("Digite uma reclamação: ");
					ocorrencias[x] = sc.nextLine();
				}
				break;
			}
			case 2: {
				boolean vazio = true;
				for (String item : ocorrencias) {
					if (item != null) {
						vazio = false;
						break;
					}
				}

				if (vazio == false) {
					System.out.println("Lista de reclamações: ");
					for (int x = 0; x < ocorrencias.length; x++) {
						System.out.println(x + 1 + " - " + ocorrencias[x]);
					}
				} else {
					System.out.println("A lista de reclamações está vazia.");
				}
				break;
			}
			case 3: {
				System.out.println("Sair. Até logo!");
				break;
			}
			default: {
				System.out.println("Opção inválida");
				break;
			}
			}

		}
		sc.close();
	}

	public static void menu() {

		System.out.print("|--------------------------------|\n");
		System.out.print("| Opção 1 - Inserir reclamações  |\n");
		System.out.print("| Opção 2 - Listar reclamações   |\n");
		System.out.print("| Opção 3 - Sair                 |\n");
		System.out.print("|--------------------------------|\n");

	}

}
